test/integration/framework/test_server.go

```diff
@@ -54,6 +54,31 @@ AwEHoUQDQgAEH6cuzP8XuD5wal6wf9M6xDljTOPLX2i8uIp/C/ASqiIGUeeKQtX0
 type TestServerSetup struct {
 	ModifyServerRunOptions func(*options.ServerRunOptions)
 	ModifyServerConfig     func(*controlplane.Config)
+
+	// RoundTrip, if set, will be called for every HTTP request going to the apiserver.
+	// It can be used for error injection.
+	RoundTrip func(transport http.RoundTripper, req *http.Request) (*http.Response, error)
```

Comment:
`StartTestServer(ctx context.Context, t testing.TB, setup TestServerSetup) (client.Interface, *rest.Config, TearDownFunc) {`

returns a rest.Config, you can copy it and create a new clientset for that with your roundtripper

```
client, config, tearDownFn := framework.StartTestServer(ctx, t, framework.TestServerSetup{})
config.Wrap(func(rt http.RoundTripper) http.RoundTripper {
				return roundTripperFunc(func(req *http.Request) (*http.Response, error) {
					authorizationHeaderValues.append(req.Header.Values("Authorization"))
					return rt.RoundTrip(req)
				})
			})

  newClient, err := client.NewForConfig(config)
        if err != nil {
                t.Fatal(err)
```

Reply:
I hadn't found `rest.Config.Wrap`. That's indeed simpler and make it possible to limit the changes to just `test/integration/util/util.go`. I've switched to that - please take another look.